src/ui_ng/lib/src/replication/replication.component.ts

```diff
@@ -0,0 +1,239 @@
+   jobsTotalPage: number;
+
+   toggleJobSearchOption = optionalSearch;
+   currentJobSearchOption: number;
+
+   @ViewChild(CreateEditRuleComponent) 
+   createEditPolicyComponent: CreateEditRuleComponent;
+
+   constructor(
+     private errorHandler: ErrorHandler,
+     private replicationService: ReplicationService,
+     private translateService: TranslateService) {
+   }
+
+   ngOnInit(): void {
+     if(!this.projectId) {
+       this.errorHandler.warning('Project ID is unset.');
+     }
+     this.currentRuleStatus = this.ruleStatus[0];
+     this.currentJobStatus  = this.jobStatus[0];
+     this.currentJobSearchOption = 0;
+     this.retrieveRules();
+   }
+
+   retrieveRules(): void {
+     toPromise<ReplicationRule[]>(this.replicationService
+         .getReplicationRules(this.projectId, this.search.ruleName))
+         .then(response=>{
+             this.changedRules = response || [];
+             if(this.changedRules && this.changedRules.length > 0) {
+               this.initSelectedId = this.changedRules[0].id || '';
+             }
+             this.rules = this.changedRules;
+             if(this.changedRules && this.changedRules.length > 0) {
+               this.search.ruleId = this.changedRules[0].id || '';
+               this.fetchReplicationJobs();
+             }
+           }
+         ).catch(error=>this.errorHandler.error(error));
+   } 
+
+   openModal(): void {
+     this.createEditPolicyComponent.openCreateEditRule(true);
+   }
+
+   openEditRule(rule: ReplicationRule) {
+     if(rule) {
+       let editable = true;
+       if(rule.enabled === 1) {
+         editable = false;
+       }
+       this.createEditPolicyComponent.openCreateEditRule(editable, rule.id);
+     }
+   }
+
+   fetchReplicationJobs(state?: State) { 
+     if(state && state.page) {
+       if(!state.page.to && state.page.to !== 0) {
+         return;
+       }
+       this.search.page = state.page.to + 1;
+     }
+     let params: RequestQueryParams = new RequestQueryParams();
+     params.set('status', this.search.status);
+     params.set('repository', this.search.repoName);
+     params.set('start_time', this.search.startTimestamp);
+     params.set('end_time', this.search.endTimestamp);
+
+     toPromise<ReplicationJob[]>(this.replicationService
+       .getJobs(this.search.ruleId, params))
+       .then(
+         response=>{
+           this.jobs = response;   
+         }).catch(error=>this.errorHandler.error(error));
+   }
+
+   selectOneRule(rule: ReplicationRule) {
+     if (rule) {
+       this.search.ruleId = rule.id || '';
+       this.search.repoName = '';
+       this.search.status = '';
+       this.currentJobSearchOption = 0;
+       this.currentJobStatus = { 'key': 'all', 'description': 'REPLICATION.ALL' };
+       this.fetchReplicationJobs();
+     }
+   }
+   
+   doSearchRules(ruleName: string) {
+     this.search.ruleName = ruleName;
+     this.retrieveRules();
+   }
+
+   doFilterRuleStatus($event: any) {
```

Comment:
Does rule list enable pagination? If not, I think we should enable pagination for all the datagrid.
For small scale item list, we can enable with front-end pagination. Otherwise, use server driven pagination.